shotover-proxy/src/server.rs

```diff
@@ -224,7 +249,7 @@ impl<C: Codec + 'static> TcpCodecListener<C> {
                         error!(cause = ?err, "connection error");
                     }
                 }
-                .instrument(tracing::info_span!(
+                .instrument(tracing::debug_span!(
```

Comment:
Ah, what I had was wrong, however debug_span is not what we want either.
This would cause the span to not be used when we are on the default info level.

Filtering for debug spans works the same as logs.
So info_span will only be included when we are on debug, trace or info levels.

However unlike logs, spans dont report their level.
So this means the correct span level we should use to ensure it is always used is: error_span